lib/constants/cycles.ts

```diff
@@ -1,58 +1,94 @@
+   * celebrated on the very same date each year (no matter what the day of the week),
+   * including Christmas and all the saints' days.
+   */
   ProperOfSaints: 'PROPER_OF_SAINTS',
 } as const;
 
 /**
- * Sundays cycle that can be used as metadata for liturgical days.
+ * A dynamically generated array from {@link ProperCycles} that contains all the possible proper cycles,
+ * that compose the liturgical year in the Roman rite.
  */
-export const SundaysCycles = {
+export const PROPER_CYCLES = Object.values(ProperCycles);
+
+/**
+ * Represent one of the two main cycles (available from {@link ProperCycles}),
+ * that compose the liturgical year in the Roman rite.
+ */
+export type ProperCycle = typeof PROPER_CYCLES[number];
+
+/**
+ * A three-year cycles for the Sunday mass readings (and some solemnities).
+ * The years are designated A, B, or C. Each yearly cycle begins on the first Sunday of Advent.
+ * Year B follows year A, year C follows year B, then back again to A.
+ */
+export const SundayCycles = {
   YearA: 'YEAR_A',
   YearB: 'YEAR_B',
   YearC: 'YEAR_C',
 } as const;
 
 /**
- * Weekdays cycle that can be used as metadata for liturgical days.
+ * A dynamically generated array from {@link SundayCycles} that contains all the possible
+ * three-year Sunday cycles.
+ */
+export const SUNDAY_CYCLES = Object.values(SundayCycles);
+
+/**
+ * Represent one of the three-year Sunday cycles (available from {@link SundayCycles}).
+ */
+export type SundayCycle = typeof SUNDAY_CYCLES[number];
+
+/**
+ * A two-year cycle for the weekday mass readings (also called Cycle I and Cycle II).
+ * Odd-numbered years are the year 1 (or Cycle I); even-numbered ones are the year 2 (or Cycle II).
  */
-export const WeekdaysCycles = {
+export const WeekdayCycles = {
   Year1: 'YEAR_1',
   Year2: 'YEAR_2',
 } as const;
 
 /**
- * Psalter weeks that can be used as metadata for liturgical days.
+ * A dynamically generated array from {@link WeekdayCycles} that contains all the possible
+ * two-year weekday cycles.
  */
-export const PsalterWeeksCycles = {
-  Week1: 'WEEK_1',
-  Week2: 'WEEK_2',
-  Week3: 'WEEK_3',
-  Week4: 'WEEK_4',
-} as const;
+export const WEEKDAY_CYCLES = Object.values(WeekdayCycles);
 
 /**
- * A dynamically generated constant consisting of all the enum keys in [[LITURGICAL_DAY_CYCLES]]
+ * Represent one of the two-year weekday cycles (available from {@link WeekdayCycles}).
  */
-export const PROPER_CYCLE = Object.values(ProperCycles);
-export type ProperCycle = typeof PROPER_CYCLE[number];
+export type WeekdayCycle = typeof WEEKDAY_CYCLES[number];
 
 /**
- * A dynamically generated constant consisting of all the enum keys in [[SUNDAY_CYCLES]]
+ The four-week cycle of the psalter is coordinated with the liturgical year in such a way that
+ on the First Sunday of Advent, the First Sunday in Ordinary Time, the First Sunday of Lent,
+ and Easter Sunday the cycle is always begun again with Week 1 (others being omitted when necessary).
```

Comment:
```suggestion
 The psalter is divided into four weeks which are coordinated with the liturgical year in such a way that
 on the First Sunday of Advent, the First Sunday in Ordinary Time, the First Sunday of Lent,
 and Easter Sunday the cycle is always begins with Week 1 (others being omitted when necessary).
```

Reply:
This text is coming from the [General Instructions of the Liturgy of the Hours](https://divineoffice.org/general-instructions/), § 133.
In my opinion, at least for the first line, the general instruction remains more precise.

However, I forgot to mention the reference (§ 133) and I should add it.

Reply:
Okay, now I made a more in-depth search on `one cycle lasting three years` vs `three cycles lasting a year`. It seems that either:

1. I take everything too literally, but the common usage is less literal (i.e. `a three-year cycle` actually means `three year-long cycles`); or
2. there is actually only one, three-year-long cycle devided into three ‘sections’ by year.

I am not sure which one is right, but `three/two-year cycle` phrase is actually used within official Church documents (and it does not matter in the context of romcal), therefore should be kept.

> However, I forgot to mention the reference (§ 133) and I should add it.

It might be a good idea to add such references to each and every code comment which cites a portion of it. However, I suggest to use a full citation (e.g. `General Instructions of the Liturgy of the Hours, I, § 133) every time we cite a particular document. Maybe we could use some shortcuts (e.g. `GILH` instead of `General Instructions of the Liturgy of the Hours`), but than it might be a good idea to have a list of them in the docs regardless if they are common or not.

Reply:
Original suggestion + the reference:
```suggestion
 The four-week cycle of the psalter is coordinated with the liturgical year in such a way that
 on the First Sunday of Advent, the First Sunday in Ordinary Time, the First Sunday of Lent,
 and Easter Sunday the cycle is always begun again with Week 1 (others being omitted when necessary)
 (GILH #133).
```

Reply:
I've added the `GILH` acronym in the main `README.md`

Reply:
What about moving the reference at the beginning of the comment? Also I suggest to use square brackets and section sign.

```suggestion
[GILH §133] The four-week cycle of the psalter is coordinated with the liturgical year in such a way that
 on the First Sunday of Advent, the First Sunday in Ordinary Time, the First Sunday of Lent,
 and Easter Sunday the cycle is always begun again with Week 1 (others being omitted when necessary).
```